libtransmission/net.h

```diff
@@ -340,7 +340,7 @@ struct tr_socket_address
         return address_;
     }
 
-    [[nodiscard]] constexpr auto port() const noexcept
+    [[nodiscard]] constexpr auto& port() const noexcept
```

Comment:
This is a style issue and maybe there's no objectively right or wrong answer to this, but for _very_ cheap-to-copy objects like `tr_port`, returning by value seems to have less downside.

I was thinking of this as a loose extension of the "in" parameters guideline F.16 from the CppCoreGuidelines:

> F.16: For "in" parameters, pass cheaply-copied types by value and others by reference to const
>
> What is "cheap to copy" depends on the machine architecture, but two or three words (doubles, pointers, references) are usually best passed by value. When copying is cheap, nothing beats the simplicity and safety of copying, and for small objects (up to two or three words) it is also faster than passing by reference because it does not require an extra indirection to access from the function.

In fairness, this rule applies to "in" parameters rather than return values; however, I think the same rationale works for return values.

Reply:
50470c2